src/Couchbase.Lite.Shared/Query/NQuery.cs

```diff
@@ -0,0 +1,155 @@
+﻿// 
+//  NQuery.cs
+// 
+//  Copyright (c) 2017 Couchbase, Inc All rights reserved.
+// 
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+// 
+//  http://www.apache.org/licenses/LICENSE-2.0
+// 
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+// 
+
+using Couchbase.Lite.Query;
+using Couchbase.Lite.Support;
+using JetBrains.Annotations;
+using LiteCore.Interop;
+using System;
+using System.Collections.Generic;
+
+namespace Couchbase.Lite.Internal.Query
+{
+    internal sealed class NQuery : QueryBase
+    {
+        #region Variables
+        private const string Tag = nameof(NQuery);
+        private string _n1qlQueryExpression = "";
+        #endregion
+
+        #region Properties
+        [NotNull]
+        internal ThreadSafety ThreadSafety { get; } = new ThreadSafety();
+        #endregion
+
+        #region Constructors
+        public NQuery(string n1qlQueryExpression, Database database) : base()
+        {
+            Database = database;
+            _n1qlQueryExpression = n1qlQueryExpression;
+        }
+        #endregion
+
+        #region Override Methods
+        public override unsafe IResultSet Execute()
+        {
+            if (Database == null) {
+                throw new InvalidOperationException(CouchbaseLiteErrorMessage.InvalidQueryDBNull);
+            }
+
+            var options = C4QueryOptions.Default;
+            var paramN1ql = _n1qlQueryExpression.FLEncode();
+
+            var e = (C4QueryEnumerator*)ThreadSafety.DoLockedBridge(err =>
+            {
+                if (_disposalWatchdog.IsDisposed) {
+                    return null;
+                }
+
+                if (_c4Query == null) {
+                    Check();
+                }
+
+                var localOpts = options;
+                return NativeRaw.c4query_run(_c4Query, &localOpts, FLSlice.Null, err);
+            });
+
+            paramN1ql.Dispose();
+
+            if (e == null) {
+                return new NullResultSet();
+            }
+
+            var retVal = new QueryResultSet(this, ThreadSafety, e, _columnNames);
+            _history.Add(retVal);
+            return retVal;
+        }
+
+        public override string Explain()
+        {
+            return _n1qlQueryExpression;
```

Comment:
I don't think this is correct, why is it not the same as XQuery?

Reply:
revert to use c4query_explain(_c4Query) for query explain